tests/test_thrift_backend.py

```diff
@@ -968,6 +968,42 @@ def test_handle_execute_response_sets_active_op_handle(self):
 
         self.assertEqual(mock_resp.operationHandle, mock_cursor.active_op_handle)
 
+    @patch("thrift.transport.THttpClient.THttpClient")
+    @patch("databricks.sql.thrift_backend._retry_policy", new_callable=retry_policy_factory)
+    def test_make_request_will_retry_GetOperationStatus(
+            self, mock_retry_policy, t_transport_class):
+       
+        import thrift
+        from databricks.sql.thrift_api.TCLIService import TCLIService
+        from databricks.sql.exc import RequestError
+        from databricks.sql.utils import NoRetryReason
+
+        t_transport_instance = t_transport_class.return_value
+        t_transport_instance.code = 200
+        t_transport_instance.headers = {"foo": "bar"}
+
+        protocol = thrift.protocol.TBinaryProtocol.TBinaryProtocol(t_transport_class)
+        client = TCLIService.Client(protocol)
+
+        req = ttypes.TGetOperationStatusReq(
+            operationHandle=self.operation_handle,
+            getProgressUpdate=False,
+        )
+        
+        thrift_backend = ThriftBackend(
+            "foobar",
+            443,
+            "path", [],
+            _retry_stop_after_attempts_count=2,
+            _retry_delay_max=0,
+            _retry_delay_min=0)
+
+        with self.assertRaises(RequestError) as cm:
+            thrift_backend.make_request(client.GetOperationStatus, req)
+
+        self.assertEqual(NoRetryReason.OUT_OF_ATTEMPTS.value, cm.exception.context["no-retry-reason"])
```

Comment:
Prior to this change the test looked like this:
```suggestion
        self.assertEqual(NoRetryReason.NOT_RETRYABLE.value, cm.exception.context["no-retry-reason"])
```